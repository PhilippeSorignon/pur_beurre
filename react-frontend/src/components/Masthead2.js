import React from "react";

class Masthead2 extends React.Component {
  render() {
    return (
      <header
        id="masthead-2"
        style={{
          backgroundImage:
            'url("https://static.openfoodfacts.org/images/products/405/648/913/7108/front_fr.22.full.jpg")'
        }}
      >
        <div className="container h-100">
          <div className="row h-100 align-items-center justify-content-center text-center">
            <div
              className="col-lg-10 align-self-end mt-10"
              style={{
                top: "-50%"
              }}
            >
              <h1 className="text-uppercase text-white font-weight-bold">
                Fromage
              </h1>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

export default Masthead2;
