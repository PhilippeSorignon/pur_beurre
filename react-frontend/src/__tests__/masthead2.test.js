import React from 'react';
import ReactDOM from 'react-dom';
import Masthead2 from '../components/Masthead2';

import { render, cleanup } from '@testing-library/react';
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Masthead2 renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Masthead2 image="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260" heading="Nutella" />, div);
});

it("Masthead2 render correctly", () => {
  const { getByRole } = render(<Masthead2 image="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260" heading="Nutella" />);
  expect(getByRole('heading')).toHaveTextContent("Nutella");
});

it("Masthead2 matches snapshot", () => {
  const tree = renderer.create(<Masthead2 image="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260" heading="Nutella" />).toJSON();
  expect(tree).toMatchSnapshot();
})
