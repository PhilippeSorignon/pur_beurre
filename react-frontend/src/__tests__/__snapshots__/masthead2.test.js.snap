// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead2 matches snapshot 1`] = `
<header
  id="masthead-2"
  style={
    Object {
      "backgroundImage": "url(https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260)",
    }
  }
>
  <div
    className="container h-100"
  >
    <div
      className="row h-100 align-items-center justify-content-center text-center"
    >
      <div
        className="col-lg-10 align-self-end mt-10"
        style={
          Object {
            "top": "-50%",
          }
        }
      >
        <h1
          className="text-uppercase text-white font-weight-bold"
        >
          Nutella
        </h1>
      </div>
    </div>
  </div>
</header>
`;
